fontstash/fontstash.h

```diff
@@ -1567,7 +1615,13 @@ static __inline void fons__vertices(FONScontext* stash, FONSquad q, FONSstate* s
     fons__vertex(stash, q.x1, q.y1, q.s1, q.t1, state->color);
 }
 
-void fonsSetShaping(FONScontext* stash, const char* script, const char* direction, const char* language)
+void fonsInitShapingPlan(FONScontext* stash, const char* str) {
+    FONSUTF32string fons32String = stash->params.toUTF32Alloc(stash->params.userPtr, str);
+    fons__hb_initShapingPlan(stash, fons32String);
+    stash->params.UTF32Free(stash->params.userPtr, fons32String);
+}
```

Comment:
Is this really necessary? This is pretty much what hb_buffer_guess_segment_properties (hb_buffer_t *buffer) does.

(Though I was looking if one can get hb_codepoints from the hb_buffer to avoid this complicated looking utf32 conversion :)